### ISSUE TITLE
[Enhancement]: Build in Audible DRM removal

### ISSUE THREAD

**Issue by therecluse26** [advplyr/audiobookshelf#2328]
### Describe the feature/enhancement

This repo allows the user to find their Audible file's activation bytes and generate/run ffmpeg to strip Audible's DRM: https://audible-tools.kamsker.at/

It'd be fantastic if you could just bundle this in as a feature so that you can just directly import an Audible audiobook file and have Audiobookshelf strip the DRM during the upload process.

**Comment by emigrating** [comment]
I personally use Libation for my Audible backups and don't really see why a media _server_ software should offer the same functionality.

I mean, sure, it's very easy to get around the DRM Audible utilizes, but this is just scope creep IMO and you end up with a jack of all trades, master of none kind of product. AFAIK, neither Plex nor Jellyfin (or any of the other servers) offer a way for you to rip your BDs natively?

Just my $0.02 worth though.

**Comment by therecluse26** [comment]
@emigrating  I disagree. I think that one of the most common use cases for using something like Audiobookshelf is to regain true ownership over purchased Audible books, and I'd bet that a substantial portion of this app's user base has to go through the painful task of manually ripping DRM from their Audible files before uploading. I think it's unfair to characterize a convenience feature that would benefit a large portion of your users as "scope creep," especially when the code for the feature already exists.

If I made a PR to integrate this feature, would it be rejected?

**Comment by nichwall** [comment]
There are currently no plans for DRM removal to be included within ABS. There are already a number of other tools which do this.

It could be a plugin (like how Calibre does it), but there is not a timetable on when plugin support will be added

**Comment by therecluse26** [comment]
One last thing on the "scope creep" point. Audiobookshelf has e-book support. That feels like a much larger instance of scope creep than what I'm suggesting.

**Comment by Sapd** [comment]
Btw. a feature like this can bring the developer, the project owner and possibly users into a difficult legal position.
For example in JDownloader some random guy did a PR for something like this, and their CI build it; JDownloader was then sued and had to pay a fine.

See also https://www.linux-magazin.de/ausgaben/2014/04/recht/ (article in German).

I would also advise everyone developing such a feature, to check the laws in their respective country.
